documentation/External-Grafana/ReadMe.md

```diff
@@ -0,0 +1,179 @@
+- Adding remote data sources such as Amazon Managed Service for Prometheus.
+- Creating Dashboards.
+- Setting up alerting.
+- And Many more.
+
+## Proposal
+
+In short the proposal in this document is about enhancing the Grafana Operator to support the integration to managed grafana services. We would need to enhance the current version of Grafana Operator to support the following :
+
+- Defining external Grafana instance as Grafana source to Grafana operator.
+- Creating Grafana dashboards on external Grafana instance.
+- Adding data sources to external Grafana instance.
+- Creating alerting on external Grafana instance.
+
+### Defining external Grafana instance as Grafana source to Grafana operator.
+
+Today Grafana operator supports self managed grafana instance as a Grafana source to Grafana operator. With this new feature, we would like to add a new CRD by name `GrafanaInstance` which basically creates an identity to an external Grafana instance. 
+
+> CRD design of how a `GrafanaInstance` will look like. In this sample design, `url` should be preloaded in a ConfigMap and `grafana_api_key` should be loaded to a Secret. Choice of using external secrets or loading secrets manually is end user responsibility :
+
+```.yaml
+apiVersion: integreatly.org/v1alpha1
+kind: GrafanaInstance
+metadata:
+  name: amazon-managed-grafana
+spec:
+  external:
+    url: <type ConfigMapSelector>
+    grafana_api_key: <type SecretKeySelector>
+```
+
+> CRD design for `GrafanaInstance` with a default setup for an external Grafana Instance. With this all Grafana Operator operations such as `GrafanaDataSource`, `GrafanaDashboard` will point to this default instance :
+
+```.yaml
+apiVersion: integreatly.org/v1alpha1
+kind: GrafanaInstance
+metadata:
+  name: amazon-managed-grafana
+  annotations:
+    grafanainstance.kubernetes.io/is-default-grafana: "true"
+spec:
+  external:
+    url: <type ConfigMapSelector>
+    grafana_api_key: <type SecretKeySelector>
+```
+
+### Creating Grafana dashboards on Amazon Managed Grafana.
+
+Today Grafana operator supports the creation of Grafana dashboards on self managed grafana instance in the cluster where the Grafana operator is installed. Grafana operator should be enhanced to support creating dashboards on external Grafana instance. 
+
+> In order to accomplish this,  `GrafanaDashboard` CRD should be modified to use `GrafanaInstance` as an input parameter to point to external Grafana instance. This allows customers to use GitOps approach to create Grafana Dashboards on remote Grafana instances.
+
+```.yaml
+apiVersion: integreatly.org/v1alpha1
+kind: GrafanaDashboard
+metadata: 
+  name: grafana-dashboard-from-url
+  labels: 
+    app: grafana
+spec:
+  grafanainstance: "amazon-managed-grafana"
+  url: https://raw.githubusercontent.com/integr8ly/grafana-operator/master/deploy/examples/remote/grafana-dashboard.json
+```
+
+### Adding data sources to external Grafana instance.
```

Comment:
This will be solved due to the label selectors that we already have in GrafanaDatasource so this can be removed.
We will solve this with the standard Grafana kind as mentioned earlier.